### PR TITLE
[bitnami/spark] Add extraVolumeClaimTemplates in master and worker StatefulSets

### LINKED ISSUE #17283: Spark - 3.3.2 - Volume for worker
### Name and Version

bitnami/spark:3.3.2

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Short explanation:
We cannot mount PVC to workers since there is not volumeClaimTemplates in worker statefulset.
So here is a test with extra volumes:

First case:
1. Create values file:
```yaml
worker:
  extraVolumes:
    - name: spark-data
      persistentVolumeClaim:
        claimName: spark-data
  extraVolumeMounts:
    - name: spark-data
      mountPath: '/opt/bitnami/spark/work'
```
2. Install with helm:
`helm -n spark upgrade --install spark-cluster bitnami/spark --version 6.6.3 -f values.yaml`

Error: spark-data PVC doesn't exist

Second case:
1. Create PVC
```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: spark-data
  namespace: spark
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 15Gi
  storageClassName: managed-csi
  volumeMode: Filesystem
```
2. Create values file:
```yaml
worker:
  extraVolumes:
    - name: spark-data
      persistentVolumeClaim:
        claimName: spark-data
  extraVolumeMounts:
    - name: spark-data
      mountPath: '/opt/bitnami/spark/work'
```
3. Install with helm:
`helm -n spark upgrade --install spark-cluster bitnami/spark --version 6.6.3 -f values.yaml`

First worker will start successfully but the second will fail because of multi-attach error.

### Are you using any custom parameters or values?

```yaml
worker:
  extraVolumes:
    - name: spark-data
      persistentVolumeClaim:
        claimName: spark-data
  extraVolumeMounts:
    - name: spark-data
      mountPath: '/opt/bitnami/spark/work'
```

### What is the expected behavior?

I want to have running 2 workers and each one should have its own volume.

### What do you see instead?

We can set the configuration for worker to have volumeClaimTemplates in statefullset.

### Additional information

With the current helm, we cannot create spark cluster with volumes.
The main problem is if we are using HDD for k8s nodes, we cannot mount SSD to spark workers.

### PR DESCRIPTION
**Pull request by ognjen-it**
### Description of the change

I just added the ability to mount volumes (PVC) to master and worker nodes

### Benefits

I we run Kubernetes nodes with HDD, Spark would be very slow. So, if we mount PVC which could be premium SSD on Azure or gp3 on AWS, Spark can have much better performance.

### Possible drawbacks

There is not drawbacks

### Applicable issues

- fixes #17283

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/spark/templates/statefulset-master.yaml [hunk 1]
```diff
@@ -344,3 +344,6 @@ spec:
             secretName: {{ template ".Values.initScriptsSecret" . }}
             defaultMode: 0755
         {{- end }}
+  {{- if .Values.worker.extraVolumeClaimTemplates }}
+  volumeClaimTemplates: {{- include "common.tplvalues.render" (dict "value" .Values.worker.extraVolumeClaimTemplates "context" $) | nindent 8 }}
+  {{- end }}
```
Comment (FraPazGal) [inline bitnami/spark/templates/statefulset-master.yaml:349]: I think you forgot to modify this, the value used should be `.Values.master.xxxx`:

```suggestion
  {{- if .Values.master.extraVolumeClaimTemplates }}
  volumeClaimTemplates: {{- include "common.tplvalues.render" (dict "value" .Values.master.extraVolumeClaimTemplates "context" $) | nindent 8 }}
  {{- end }}
```

File: bitnami/spark/README.md
Comment (FraPazGal) [inline bitnami/spark/README.md:1]: You have added the entries to the README and templates but not the `values.yaml` file, could you please double-check this?

**Review by FraPazGal — COMMENTED** [review]
Hi @ognjen-it, thanks a lot for the PR! I have added a couple of comments, could you please review them?

File: bitnami/spark/templates/statefulset-master.yaml (hunk 1 above)
Comment (ognjen-it) [reply bitnami/spark/templates/statefulset-master.yaml]: Thank you for review. When I sow an email and changed it without looking at Git Hub, so now I could not Sign off and commit suggestion that you provided. Sorry.
However, I changed it as you suggested.

**Comment by FraPazGal** [comment]
The PR looks good to me. It seems there is a conflict on `Chart.yaml`, @ognjen-it could you resolve it before I approve & merge this?

**Comment by ognjen-it** [comment]
I really hate rebase. Let me open new PR :)

**Comment by ognjen-it** [comment]
#17353

**Pull request closed without merging**